README_python.md

```diff
@@ -289,6 +552,201 @@ But if you construct the numpy arrays yourself (like above), you can pass
 passing in an array constructed somewhere else, the easiest thing to do is to
 `.transpose()` it before passing it to your Halide code.)
 
+### Advanced Generator-Related Topics
+
+#### Generator Aliases
+
+A Generator alias is a way to associate a Generator with one (or more) specific
+sets of GeneratorParams; the 'alias' is just another registered name. This
+offers a convenient alternative to specifying multiple sets of GeneratorParams
+via the build system. To define alias(es) for a Generator, just add the
+`@hl.alias` decorator before `@hl.generator` decorator:
+
+```
+@hl.alias(
+    xor_generator={"op": "xor"},
+    and_generator={"op": "and"},
+    or_generator={"op": "or"}
+)
+@hl.generator("logical_op_generator")
+class LogicalOpGenerator:
+    ...
+```
+
+#### Dynamic Inputs and Outputs
+
+If you need to build `Input` and/or `Output` dynamically, you can define a
+`configure()` method. It will always be called after all `GeneratorParam` values
+are valid, but before `generate()` is called. Let's take our example and add an
+option to pass an offset to be added after the logical operator is done:
+
+```
+import halide as hl
+
+x = hl.Var('x')
+y = hl.Var('y')
+
+_operators = {
+  'xor': lambda a, b: a ^ b,
+  'and': lambda a, b: a & b,
+  'or':  lambda a, b: a | b
+}
+
+# Apply a mask value to a 2D image using a logical operator that is selected at compile-time.
+@hl.generator(name = "logical_op_generator")
+class LogicalOpGenerator:
+    op = hl.GeneratorParam("xor")
+    with_offset = hl.GeneratorParam(False)
+
+    input = hl.InputBuffer(hl.UInt(8), 2)
+    mask = hl.InputScalar(hl.UInt(8))
+
+    output = hl.OutputBuffer(hl.UInt(8), 2)
+
+    def configure(g):
```

Comment:
Do we want to use the `g = self` idiom here? Maybe I missed it, but if we want to advocate for using `g` in place of `self` for that argument name, we should call it out and indicate that it's just for terseness and isn't magic.

Reply:
Yep, will call this out earlier and more emphatically

Reply:
Edit: after looking at this some I wonder if this is a mistake in our style thoughts -- if you didn't notice the docs on a read-through, other folks won't either, and will wonder WTF is going on. Maybe it would be clearer to have the convention (in our code at least) be

```
    def generate(self):
        g = self  # for terseness in expressions
       ...
```

? 

(I won't make any changes here till I hear your thoughts)

Reply:
Any thoughts on this one?